src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -94,3 +100,56 @@ struct NormalDotNumericalFlux : db::DataBoxPrefix {
   static constexpr db::DataBoxString_t label = "NormalDotNumericalFlux";
 };
 }  // namespace Tags
+
+namespace Prefixes_detail {
+template <typename, template <typename...> class Prefix, typename Tag,
+          typename... Args>
+struct add_tag_prefix_impl;
+
+template <template <typename...> class Prefix, typename Tag, typename... Args>
+struct add_tag_prefix_impl<std::false_type, Prefix, Tag, Args...> {
+  using type = Prefix<Tag, Args...>;
+};
+
+template <template <typename...> class Prefix, typename Tag, typename... Args>
+struct add_tag_prefix_impl<std::true_type, Prefix, Tag, Args...> {
+  using type = Prefix<
+      Tags::Variables<tmpl::transform<typename Tag::tags_list,
+                                      tmpl::bind<Prefix, tmpl::_1, Args...>>>,
+      Args...>;
+};
+
+template <typename>
+struct remove_tag_prefix_impl;
+
+template <typename UnprefixedTag, template <typename...> class Prefix,
+          typename... Args>
+struct remove_tag_prefix_impl<Prefix<UnprefixedTag, Args...>> {
+  static_assert(cpp17::is_base_of_v<db::DataBoxTag, UnprefixedTag>,
+                "Unwrapped tag is not a DataBoxTag");
+  using type = UnprefixedTag;
+};
+
+template <typename... VariablesTags, template <typename...> class Prefix,
+          typename... Args>
+struct remove_tag_prefix_impl<
+    Prefix<Tags::Variables<tmpl::list<VariablesTags...>>, Args...>> {
+  using type =
+      Tags::Variables<tmpl::transform<tmpl::list<VariablesTags...>,
+                                      remove_tag_prefix_impl<tmpl::_1>>>;
+};
```

Comment:
These will be really useful, a minor change to the implementation:

```cpp
template <bool IsVariables>
struct add_tag_prefix_impl;

template <>
struct add_tag_prefix_impl<false> {
  template <template <typename...> class Prefix, typename Tag, typename... Args>
  using f = Prefix<Tag, Args...>;
};

template <>
struct add_tag_prefix_impl<true> {
  template <template <typename...> class Prefix, typename Tag, typename... Args>
  using f = Prefix<
      Tags::Variables<tmpl::transform<typename Tag::tags_list,
                                      tmpl::bind<Prefix, tmpl::_1, Args...>>>,
      Args...>;
};
```
This separates the algorithm from the data so fewer structs are created. Type aliases are cheaper to evaluate. I also prefer to name the template parameter used on the struct so that the condition is clear. In this case when `IsVariables` is true, we know we are dealing with a Variables class.

```cpp
template <typename>
struct remove_tag_prefix_impl;

template <typename UnprefixedTag, template <typename...> class Prefix,
          typename... Args>
struct remove_tag_prefix_impl<Prefix<UnprefixedTag, Args...>> {
  static_assert(cpp17::is_base_of_v<db::DataBoxTag, UnprefixedTag>,
                "Unwrapped tag is not a DataBoxTag");
  using type = UnprefixedTag;
};

template <typename... VariablesTags, template <typename...> class Prefix,
          typename... Args>
struct remove_tag_prefix_impl<
    Prefix<Tags::Variables<tmpl::list<VariablesTags...>>, Args...>> {
  using type = Tags::Variables<
      tmpl::list<typename remove_tag_prefix_impl<VariablesTags>::type...>>;
};
```
The only change is getting rid of the `transform` with just a straight pack expansion to avoid an additional layer of type creation.